compiler/passes/codegen.cpp

```diff
@@ -1100,9 +1100,16 @@ static void codegen_header(bool isHeader) {
   genComment("Virtual Method Table");
   genVirtualMethodTable(types,isHeader);
 
-  genComment("Global Variables");
-  forv_Vec(VarSymbol, varSymbol, globals) {
-    varSymbol->codegenGlobalDef(isHeader);
+  if(fIncrementalCompilation) {
+    genComment("Global Variables");
+    forv_Vec(VarSymbol, varSymbol, globals) {
+      varSymbol->codegenGlobalDef(isHeader);
+    }
+  } else if(isHeader) {
+      genComment("Global Variables");
+      forv_Vec(VarSymbol, varSymbol, globals) {
+        varSymbol->codegenGlobalDef(isHeader);
+      }
```

Comment:
This looks like it is doing the exact same thing in both branches, would condensing that it into a single branch be appropriate?